scripts/funcotator/data_sources/getDbSNP.sh

```diff
@@ -14,211 +14,213 @@ MINARGS=0
-        rm ${md5File}
-    fi
-
-    echo "${indentSpace}Creating Config File ... "
-    createConfigFile "${DATA_SOURCE_NAME}" "${version}" ${filePrefix}_${vcfFile} "ftp://ftp.ncbi.nih.gov/snp/organisms/${remoteFolder}/VCF/${vcfFile}" > ${outputFolder}/${DATA_SOURCE_NAME}.config
+	local remoteFolder=$1
+	local outputFolder=$2
+	local filePrefix=$3
+
+	local listingFile=$( makeTemp )
+	local indentSpace="    "
+	local version=$( echo "${remoteFolder}" | sed "s#.*human_\\(.*b${BUILD_NUMBER}\\).*#\\1#g" )
+
+	local tmpVcfFile=$( makeTemp )
+
+	curl ftp://ftp.ncbi.nih.gov/snp/organisms/${remoteFolder}/VCF/ 2>/dev/null > ${listingFile}
+
+	vcfFile=$( cat ${listingFile} | awk '{print $9}' | grep -E "${SRC_FILE_REGEX}" )
+	tbiFile=$( cat ${listingFile} | awk '{print $9}' | grep -E "${TBI_FILE_REGEX}" )
+	md5File=$( cat ${listingFile} | awk '{print $9}' | grep -E "${MD5_FILE_REGEX}" )
+
+	echo "${indentSpace}Retrieving MD5 sum file: ftp://ftp.ncbi.nih.gov/snp/organisms/${remoteFolder}/VCF/${md5File} ... "
+	wget ftp://ftp.ncbi.nih.gov/snp/organisms/${remoteFolder}/VCF/${md5File}
+
+	# Get the VCF file
+	echo "${indentSpace}Retrieving VCF file: ftp://ftp.ncbi.nih.gov/snp/organisms/${remoteFolder}/VCF/${vcfFile} ... "
+	wget ftp://ftp.ncbi.nih.gov/snp/organisms/${remoteFolder}/VCF/${vcfFile}
+
+	#echo "${indentSpace}Retrieving VCF Index file: ftp://ftp.ncbi.nih.gov/snp/organisms/${remoteFolder}/VCF/${tbiFile} ... "
+	#wget ftp://ftp.ncbi.nih.gov/snp/organisms/${remoteFolder}/VCF/${tbiFile}
+
+	# We can only verify the checksum with hg38 because we modify the hg19 file as we stream it in:
+	echo "${indentSpace}Verifying VCF checksum ..."
+	if [[ "$(uname)" == "Darwin" ]] ; then
+		which md5sum-lite &> /dev/null
+		r=$?
+		if [ $r == 0 ] ; then
+			checksum=$( md5sum-lite ${vcfFile} | awk '{print $1}' | sed -e 's#^[ \t]*##g' -e 's#[ \t]*$##g' )
+			expected=$( head -n1 ${md5File} | awk '{print $1}' | sed -e 's#^[ \t]*##g' -e 's#[ \t]*$##g' )
+
+			if [[ "${checksum}" != "${expected}" ]] ; then
+				error "DOWNLOADED FILE IS CORRUPT!  (${checksum} != ${expected})"
+				error "FAILING"
+				exit 4
+			fi
+		else
+			error "Unable to validate md5sum of file: cannot locate 'md5sum-lite'.  Use these data with caution."
+		fi
+	else
+		which md5sum &> /dev/null
+		r=$?
+		if [ $r == 0 ] ; then
+			checksum=$( md5sum ${vcfFile} | awk '{print $1}' | sed -e 's#^[ \t]*##g' -e 's#[ \t]*$##g' )
+			expected=$( head -n1 ${md5File} | awk '{print $1}' | sed -e 's#^[ \t]*##g' -e 's#[ \t]*$##g' )
+
+			if [[ "${checksum}" != "${expected}" ]] ; then
+				error "DOWNLOADED FILE IS CORRUPT!  (${checksum} != ${expected})"
+				error "FAILING"
+				exit 4
+			fi
+		else
+			error "Unable to validate md5sum of file: cannot locate 'md5sum'.  Use these data with caution."
+		fi
+	fi
+
+	# Now change the contigs in the file:
+	cat ${vcfFile} | gunzip | sed -e 's#^\([0-9][0-9]*\)#chr\1#' -e 's#^MT#chrM#' -e 's#^X#chrX#' -e 's#^Y#chrY#' | bgzip > ${tmpVcfFile} 
```

Comment:
`sed -i` behaves differently on linux and OSX so I try to avoid using it in scripts.  I can change that I guess - it'll just make the code more complicated.

Yeah - I misunderstood what you meant before.  I'll add in anchors to the regex to ensure that only the correct contigs are changed.